src/cordova/test.action.mjs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from 'url';
+import os from 'os';
+
+import cordovaLib from 'cordova-lib';
+const {cordova} = cordovaLib;
+
+import {execSync} from 'child_process';
+import {getRootDir} from '../build/get_root_dir.mjs';
+import path from 'path';
+import {getTestParameters} from './get_test_parameters.mjs';
+
+/**
+ * @description Tests the parameterized cordova binary (ios, macos).
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {platform: cordovaPlatform, osVersion, deviceModel, cpuArchitecture} = getTestParameters(parameters);
+  const outlinePlatform = cordovaPlatform === 'osx' ? 'macos' : cordovaPlatform;
+
+  console.log(`Testing OutlineAppleLib on ${outlinePlatform}, ${osVersion}`);
+
+  if (outlinePlatform === 'macos' || outlinePlatform === 'ios') {
+    if (os.platform() !== 'darwin') {
+      throw new Error('Building an Apple binary requires xcodebuild and can only be done on MacOS');
+    }
+
+    const PACKAGE_PATH = path.join(getRootDir(), '/src/cordova/apple/OutlineAppleLib/');
+    const PACKAGE_NAME = `OutlineAppleLib`;
+
+    if (outlinePlatform === 'macos') {
+      let xcodeDestination = `platform=macOS,arch=${cpuArchitecture}`;
+
+      if (osVersion) {
+        xcodeDestination += `,OS=${osVersion}`;
+      }
+
+      execSync(`xcodebuild test -scheme "${PACKAGE_NAME}" -destination "${xcodeDestination}"`, {
+        cwd: PACKAGE_PATH,
+        stdio: 'inherit',
+      });
+    }
+
+    if (outlinePlatform === 'ios') {
+      let xcodeDestination = `platform=iOS Simulator,name=${deviceModel}`;
```

Comment:
Same comment. We should perhaps hardcode or default this to an iPhone with the minumum OS we support.

Reply:
@daniellacosse that doesn't give us the min version.
@eacrh we are overthinking this. Just hardcode the number. We rarely change that, and it will break if do we upgrade the code and the test no longer supports it.